app/controllers/admin/settings_controller.rb

```diff
@@ -48,6 +48,7 @@ class SettingsController < ApplicationController
     end
 
     def show
+      @guessed_host = request.host_with_port.dup
```

Comment:
Since this is only needed for the general settings page, please move it to the ` GeneralSettingsController`. You can overwrite the `show` method of the parent controller there, and simply copy what you have written here.